FirebaseFunctions/Sources/Internal/FunctionsSerializer.swift

```diff
@@ -20,21 +20,21 @@ private enum Constants {
   static let dateType = "type.googleapis.com/google.protobuf.Timestamp"
 }
 
-enum SerializerError: Error {
-  // TODO: Add parameters class name and value
-  case unsupportedType // (className: String, value: AnyObject)
-  case unknownNumberType(charValue: String, number: NSNumber)
-  case invalidValueForType(value: String, requestedType: String)
+extension FUNSerializer {
+  enum Error: Swift.Error {
+    case unsupportedType(typeName: String)
```

Comment:
I didn’t include the original object as an associated value because whoever is calling the method already has it, so it makes little sense to just forward the object as is. Please correct me if I’m wrong.

Reply:
I agree.